### PR TITLE
Transfer experiment

### PR DESCRIPTION
**Pull request by galshachaf**
for reference:
https://arxiv.org/pdf/1411.1792.pdf

### CODE REVIEW

File: analyze_model.py [hunk 1]
```diff
@@ -144,3 +144,37 @@ def generate_robustness_winnery_correlation_graph(pruned_robustness,
         pyplot.title("Robustness & Winning Ticket Intersection by Layer", loc='right', fontsize=12, fontweight=0, color='orange')
         pyplot.xlabel("Layer")
         pyplot.savefig(os.path.join(output_dir, filename), format='png')
+
+    @staticmethod
+    def generate_transfer_graph(results: dict, output_dir: str, filename="transfer_fig", base_name="base"):
+        print(results)
+        pyplot.style.use('seaborn-darkgrid')
+        palette = pyplot.get_cmap('Set1')
```
Comment (dorimedini) [inline analyze_model.py:154]: Maybe call `pyplot.figure()` here, so subsequent calls to `generate_transfer_graph` don't output the results to the same graph

File: transfer.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                 split_labels=None, *args, **kwargs):
+        if split_labels is None:
+            split_labels = sample(list(range(n_classes)), n_classes // 2)
+        data_a = SplitDataset(dataset, split_labels)
+        data_b = SplitDataset(dataset, [i for i in range(n_classes) if i not in split_labels])
+        trainer_a = trainer_class(dataset=data_a, **trainer_kwargs)
+        trainer_b = trainer_class(dataset=data_b, **trainer_kwargs)
+        super(TransferExperiment, self).__init__(model_names=["a", "b"],
+                                                 trainers={
+                                                     "a": trainer_a,
```
Comment (dorimedini) [inline transfer.py]: Maybe more descriptive model names...? Calling `self.open_model` may initiate training, and then the model is saved using the `model_name` argument (the output filename will be `a.h5`)

In any case you use the strings `"a","b"` at least twice so I would put the model names in named variables

**Review by dorimedini — CHANGES_REQUESTED** [review]

File: analyze_model.py (hunk 1 above)
Comment (galshachaf) [reply analyze_model.py:154]: why does it matter where I call it? because of get_cmap? @dorimedini

File: transfer.py (hunk 2 above)
Comment (galshachaf) [reply transfer.py]: it's descriptive once you read the paper :) I'll added a variable. @dorimedini

File: analyze_model.py (hunk 1 above)
Comment (dorimedini) [reply analyze_model.py:154]: @galshachaf I don't think you actually need to call it if you're just outputting a single plot. But `generate_transfer_graph` may be called more than once in a single experiment, and if that happens then all the results of different calls will be output to the same graph. I had the problem and adding `pyplot.figure()` to the method solved the issue, I could then call my method multiple times and get separate graphs.
If this works as is: no problem. Just test and make sure calling `generate_transfer_graph` twice makes two graphs

Comment (galshachaf) [reply analyze_model.py:154]: did you notice it's five lines down in the code?

**Comment by galshachaf** [comment]
@dorimedini now it's your turn

File: analyze_model.py (hunk 1 above)
Comment (dorimedini) [reply analyze_model.py:154]: @galshachaf hhhhh whoops

**Comment by dorimedini** [comment]
> @dorimedini now it's your turn

@galshachaf Done :)

**Review by dorimedini — APPROVED** [review]

**Pull request merged**